src/simulator.ts

```diff
@@ -88,13 +89,51 @@ export class Simulator {
 
   public async cancel() {
     this.cancelToken = true;
+    await this.delay(1000);
+  }
+
+  public async telemetry(eventName: string, status: boolean, properties?: { [key: string]: string; }) {
+    const iotHubConnectionString = status ? await Utility.getConnectionString(
+      Constants.IotHubConnectionStringKey,
+      Constants.IotHubConnectionStringTitle,
+      false,
+    ) : undefined;
+    if (eventName === Constants.SimulatorLaunchEvent) {
+      TelemetryClient.sendEvent(eventName, {
+        status: status ? "Succeeded" : "Failed",
+        reason: status ? undefined : properties.error,
+        quitWhenProcessing: this.isProcessing() ? "true" : "false",
+      }, iotHubConnectionString);
+    } else if (eventName === Constants.SimulatorSendEvent) {
+      if (status) {
+        TelemetryClient.sendEvent(eventName, {
+          status: "Succeeded",
+          devices: "" + properties.deviceConnectionStrings.length,
+          numbers: "" + properties.numbers,
+          interval: "" + properties.interval,
+          messageBody: "" + properties.messageBody,
+        }, iotHubConnectionString);
+      } else {
+        TelemetryClient.sendEvent(eventName, {
+          status: "Failed",
+          reason: "" + properties.reason,
+        }, iotHubConnectionString);
+      }
+    } else if (eventName === Constants.SimulatorCloseEvent) {
+        TelemetryClient.sendEvent(eventName, {
+          status: status ? "Succeeded" : "Failed",
+          reload: properties.reload,
```

Comment:
What does the reload mean?

Reply:
The `this.panel.dispose();` is caused by user's clicking on the close button, then the `reload` is `false`, and if this function is caused because user clicks on the menu of a selected device and then the WebView reloads with that device as default device, then the `reload` is `true`. In this way we can know the difference between these two actions.